### DISCUSSION TITLE
(please help) tablet does not work. Resizing game breaks windows.

### DISCUSSION THREAD

**Discussion by oliverlundstrom** [ppy/osu#29219] [Q&A]
I use opentabletdriver for the plugins.
My tablet does not work on osu lazer when I disable the in game "tablet" option and I am never going to use another driver than the one I already use. dont even dare suggest it.
Disabling precision mouse does nothing.

I also play with 1600x1050 resolution on stable, which has never ever had any issues.
Yet when I rescale lazer into that my: 
- mouse appears on my second monitor from time to time while in game
- for example, chrome on background thinks its in full screen but its only a part of the screen when I tab out or close the game
(see screenshot below)
![osulazernotwokringffs](https://github.com/user-attachments/assets/964f6ca5-6bcd-4de4-bedd-58abcda6db50)

game overall feels glitchy and nothing like stable which I personally hate. Instead of focusing on the games strengths you try to imagine new things making the focus less clear and the product overall more mediocre. I dont understand why you do that.

**Comment by oliverlundstrom** [comment]
see how the window thinks its maximized by the icon?

**Comment by oliverlundstrom** [comment]
specifically the mouse appears on the left hand monitor I have when I go to the left hand side of osu lazer.

**Comment by oliverlundstrom** [comment]
sorry for the ending I shouldnt type these while mad.

**Comment by oliverlundstrom** [comment]
[compressed-logs.zip](https://github.com/user-attachments/files/16447105/compressed-logs.zip)
heres my logs

**Comment by oliverlundstrom** [comment]
ig this is an issue but idk how to use this website and I dont want to annoy anyone :c

**Comment by Stoppedpuma** [comment]
You already opened a discussion at #29218, it's unnecessary to create more than one.

Please try to enable / disable "High Precision Input" and see if it makes any difference.

Regarding your graphics problem, can you see if setting the renderer to "OpenGL" fixes anything?

**Reply by oliverlundstrom** [reply]
as I said in my original thread it posted the darn thing accidentally with ctrl enter while trying to linebreak, im new to this site. I wanted to delete the post and said so in it. 
And as I said precision input does not matter. Same with renderer, I tried both of these before posting.

**Reply by Stoppedpuma** [reply]
> And as I said precision input does not matter. Same with renderer, I tried both of these before posting.

Either I'm blind or you didn't mention this anywhere.

I'm going to assume this is something with your graphics driver, are you using any unusual settings on your graphics driver? E.g Supersampling or scaling? To start I would recommend checking that you're on the latest driver and that your windows install is on the latest update as well.

**Reply by oliverlundstrom** [reply]
its the last sentence of the first paragraph

**Reply by oliverlundstrom** [reply]
drivers are normal but I can try to reset em tomorrow, I have never touched em but maybe they enable themselves automatically for osu lazer or something

**Reply by oliverlundstrom** [reply]
still does not explain the tablet issue which is the main problem and also stable works just fine, everything else works just fine, you really should assume less.

**Reply by oliverlundstrom** [reply]
yeah this is still fucked though

**Reply by Stoppedpuma** [reply]
> still does not explain the tablet issue which is the main problem

Are you using windows ink or anything related? Additionally is your tablet area bound to only the display you're playing osu! on? I had a similar issue at one point on windows with hawku or some other legacy driver.

> yeah this is still fucked though

Hmm, not sure? Do you have any external apps which change how windows behave? E.g Borderless Window which is an application on steam that turns fullscreen windows into borderless ones.

**Reply by oliverlundstrom** [reply]
Good questions but I unfortunately have all of that turned off, had those same issues back in the day xd (personally favored the devocub drivers)

also no borderless window app, I like playing in full screen for the latency but for some reason lazer resizes the whole screen in a weird way where all my desktop shortcuts get jumbled around like what used to happen to some old games on pc, maybe its a windows thing but I do not notice any issues like this on stable. Tried all the renderers, no difference, re installed my gpu drivers and no fix :/

**Reply by oliverlundstrom** [reply]
Oh my god I think I just realized why the built in tablet driver feels so weird. The area it has does not scale on the native resolution of the screen, but the resolution of the game, making it inconsistent with what I would use in my normal drivers. Atleast this would make some sense.

I confirmed this by setting an area and 1600x1050 resolution, the area felt way bigger than when I put the resolution at 1920x1080. 

I think the only thing that needs to be done is to make so that the tablet area in the lazer driver always scales with native resolution of the screen. It also feels like if I have opentabletdriver open while playing the game and using the in game drivers, the plugins do still take effect so there was nothing wrong with that, the reason why it felt weird was the resolution thing, ill just play on 1920 on lazer but that def could be fixed if there is time and I am correct, I could be wrong very easily though.

**Reply by oliverlundstrom** [reply]
the tablet area feels exactly the same as on desktop with the same width and forced aspect ratio as it does on the lazer driver, but when I put lazer resolution as 1600x1050, which is the same but just thinner, the tablet area is provably thinner if that makes any sense. I measured the distance it took to cross from one part of the screen from another with ruler and tested.

**Reply by oliverlundstrom** [reply]
and just to say it, this does not happen on opentabletdriver with stable.

**Reply by Stoppedpuma** [reply]
No clue what could be causing this issue as I haven't used windows in years. I doubt this is a windows thing but rather an issue on your system somewhere?

Probably not the answer you're looking for but if everything works fine in fullscreen then you can use "Screen Scaling" as a workaround for now.

![image](https://github.com/user-attachments/assets/371ebd57-6107-4114-a424-c2f2ea415432)

**Reply by oliverlundstrom** [reply]
but there is the issue of mouse appearing on the other monitor that hasnt gone away, probably something to do with how the code understands screen scaling to be tied to the game and not the native resolution of the screen, I dont know enough about coding yet to really say. 
It only happens then the screen is scaled down. Also when clicking the taskbar icon to get back into the game it kind of breaks and the mouse cursor shows up on the other screen.

**Reply by oliverlundstrom** [reply]
oh so you didnt go to sleep either I see

**Reply by oliverlundstrom** [reply]
actually no wait, I had the screen scaling on gameplay only just now and I did try those settings when first running into the issue.

**Reply by oliverlundstrom** [reply]
Yeah it still breaks on screen scaling set to gameplay.

**Reply by Stoppedpuma** [reply]
Unlikely but still possible, are you possibly bumping or moving your mouse unintentionally during gameplay somehow?

**Reply by oliverlundstrom** [reply]
also yeah my pc might just be cursed with the spirit of my elders but idk, other things similar to this application work fine while resizing, I do it sometimes a lot to get better performance out of games cuz of potato

**Reply by oliverlundstrom** [reply]
no im telling you, I tested the size of the tablet area and it differs in scale and size when you resize the window

**Reply by oliverlundstrom** [reply]
which does not happen on the normal tablet driver and imo shouldnt happen.

**Reply by oliverlundstrom** [reply]
I know its different when I literally test the two apps side by side, stable and lazer

**Reply by Stoppedpuma** [reply]
I think that's expected behaviour as the driver maps to the window size since it can't map to the display? Hard to explain unless you have experience with similar things but basically the driver reads the windows resolution state and maps to it. I think display would be possible but don't quote me on that.

**Reply by Stoppedpuma** [reply]
The "bump" thing was meant for the cursor displaying on your second display.

**Reply by oliverlundstrom** [reply]
I think the tabletdriver works how it works because you set your screen native resolution on that and changing osus resolution does not change that.

**Reply by oliverlundstrom** [reply]
ill draw an image if this is badly explained cuz it makes sense, if you know how varvalian played he has like full area but uses small window size to make the playing area smaller.

**Reply by Stoppedpuma** [reply]
Ok wait I think I understand the issue you're talking about with the driver area "becoming wider"? I thought this was an issue on my window manager but maybe not? Just to make sure, we're talking about the external driver, not the one built into osu! correct?

**Reply by oliverlundstrom** [reply]
changing stable resolution kind of zooms into the tablet area

**Reply by oliverlundstrom** [reply]
Oh I know it might be because there is no letterboxing.

**Reply by oliverlundstrom** [reply]
and sorry yes, I have a seperate tabletdriver I use for stable osu. to not complicate this, it works with the smaller resolution, lazer tablet driver does not, and it goes over the seperate tablet driver.

**Reply by oliverlundstrom** [reply]
its hard to explain but it literally is just lack of letterboxing.

**Reply by Stoppedpuma** [reply]
Is this what you're experiencing? Where the cursor seems to confine to the games window size?

[1722484118-av1.webm](https://github.com/user-attachments/assets/6a79dca0-f91d-4c81-b6f0-b80638df79e3)

**Reply by oliverlundstrom** [reply]
Yeah pretty much except the other way around where im trying to go smaller to change the aspect ratio of the game but not of the tabletdriver area.

**Reply by oliverlundstrom** [reply]
The game feels extremely weird and I am less confident that my tabletdriver plugins are working at all. I use some latency and I dont feel that at all, same with chatter from the cth 480 i have and the area feels different. I know it might sound like cap but I have been playing this game for a while (rn 7,8k rank without farming for 4 years.) and I can feel that it is completely different. When I draw it feels the same as osu but this feels different for some reason.

**Reply by Stoppedpuma** [reply]
The video I posted above seems to be an issue with my window manager, testing with another works fine so disregard that. A possible reason it feels different is frame limiter? You probably have stable at unlimited, capping it at 1k should make it feel nearly the same as lazer in "basically unlimited" mode?

**Reply by oliverlundstrom** [reply]
I have it at 8x but tested 2x and 4x on lazer and I have optimal on stable.

**Reply by Stoppedpuma** [reply]
Well lazer should have less input lag than stable when both are at 1k fps caps, are you feeling more latency in lazer?

**Reply by oliverlundstrom** [reply]
Nono, I have a latency plugin on my tabletdriver. I want that. It gives better aim control and reading for me. It does not work on lazer and yeah I have tried pretty much everything and you can not resize your window to 1600x1050 and use a tablet with that. Even when in fullscreen the cursor pops out of the side of the screen into the other screen until bugging out.

**Reply by Stoppedpuma** [reply]
Maybe one of your plugins is causing issues then? Can you try without any plugins and see if you're cursor is still going to your other screen?

**Reply by oliverlundstrom** [reply]
I actually thought that that might do something but nothing at all happened when I closed my tabletdrivers. I think lazer just says "fuck those completely, you dont need em, I know you have settings and plugins and everything but fuck you" and then gives you a dogshit way to map your tablet area.

**Reply by oliverlundstrom** [reply]
meaning the same issues without them too, even if that worked I wouldnt have my plugins etc

**Reply by oliverlundstrom** [reply]
I am not going to relearn muscle memory just because a game is poorly made. If this is never fixed I guess ill just die playing stable and bitter.

**Reply by oliverlundstrom** [reply]
the game does feel really responsive and nice, which is making me even more mad tbh xdd

**Reply by Stoppedpuma** [reply]
> a game is poorly made

I wouldn't necessarily say this, for the most part lazer works across all major operating systems quite well. This is the first time I've seen this and I'm willing to bet this is probably just an issue with your windows install as in my experience the older they are the more problems occur. For now Stable may be the better option for you but it may be worth trying again in the future! I'm about where you are with this as I haven't used windows in years and don't know what kind of fun bugs are in windows nowadays so I can't really help too much more, maybe someone else will have a solution.

**Reply by oliverlundstrom** [reply]
I hate the way society makes technology I truly do but I also love individuals like yourself, thank you for replying and trying to help! Not like there is a huge daily challenge thing on my profile I cant compete in. Its wonderful being alive in this fucking world. (this wasnt a gripe against you but more to the whole of society)

**Reply by Stoppedpuma** [reply]
If you really want to play the daily challenge a solution (not a very good one) is to temporarily disconnect your second display?

**Reply by oliverlundstrom** [reply]
hmmm true. That one fixes the size and I can deal with no plugins for a single map. Thanks!
